### ISSUE TITLE
appId supplied but not found

### ISSUE THREAD

**Issue by bistritapcv** [microsoft/WinAppDriver#1884]
I am using Selenium/Java/WinApPDriver.  I need to use it just for one spot where it enters a login and password into a windows box and won't work with Selenium.  I got this working fine with Winium but corporate won't let us use Winium as there have been no security upgrades.  So I am trying to cnvert to WinAppDriver.

I read the directions and created my pom file and all that.  When I try to start the windows driver I get "Could not start a new session. Response code 400. Message: Missing command parameter: appId ".  Doing debugging it looks like something is prefixing all my capabilities with "appium:" so tht WinAppDriver gets the capability "appium:appId" instead of just "appId".

My startup is the following.  Even if I misunderstand what appId is it should not say it is not provided:

			WindowsDriver app = null;

			DesiredCapabilities caps = new DesiredCapabilities();
			caps.setCapability("automationName", "Windows");
			caps.setCapability("app", "C:\\Windows\\System32\\notepad.exe");
			caps.setCapability("appId", "C:\\Windows\\System32\\notepad.exe");
			caps.setCapability("platformName", "");
			caps.setCapability("deviceName", "WindowsPC");
			caps.getCapabilityNames();
			app = new WindowsDriver(new URL("http://127.0.0.1:4723"), caps);

I have tried many different PlatformNames.  I tried updating to the latest appium.io but it did not help.  Here is part of my pom.  We have an AutomationParent which has all the things like Java, Selenium, etc.  So I had that dependency and added this:

<dependency>
     <groupId>io.appium</groupId>
     <artifactId>java-client</artifactId>
     <version>8.1.0</version>
</dependency>

**Comment by anunay1** [comment]
The appium version that you are using will not work. check #1880.

**Comment by bistritapcv** [comment]
It looks like most people use 1.22.3 but when I try that one from maven it can't be found:

<dependency>
 <groupId>io.appium</groupId>
 <artifactId>java-client</artifactId>
 <version>1.22.3</version>
</dependency>

**Comment by bistritapcv** [comment]
I also read that I don't even need appium if I am doing windows tests but in the documentation it says you need appium.io java which is what the driver uses.

**Comment by anunay1** [comment]
You need the Java client, you can  use winappdriver.exe as an alternative to appium.

**Comment by anunay1** [comment]
Which location are are looking for java client?

**Comment by bistritapcv** [comment]
I am hoping I am using only the driver part of appium and not appium itself.    The appium.io is in my pom file.  I have like this (which I saw in the documentation):

     <dependency>
         <groupId>io.appium</groupId>
         <artifactId>java-client</artifactId>
         <version>7.3.0</version>
     </dependency>

but please not I have tried several different versions.  The problem seems to be it appends an "appium: to the front of all my capabilities.  I have proved this stepping through the code where it adds an appium prefix to each capability.  Here is from the debugger.  It still showws desired capabilities but you can see that for capabilities it added "appium: (except for platform)

     {capabilities=[{appium:appId=C:\Windows\System32\notepad.exe, appium:automationName=Windows, appium:deviceName=WindowsPC, platformName=WINDOWS}], desiredCapabilities=Capabilities {appId: C:\Windows\System32\notepad..., automationName: Windows, deviceName: WindowsPC, platformName: WINDOWS}}

**Comment by anunay1** [comment]
Have you tried with WinappDriver.exe?

**Comment by anunay1** [comment]
You can download from https://github.com/Microsoft/WinAppDriver/releases and run winappdriver.exe from program 86 folder.

**Comment by anunay1** [comment]
I don't know what you mean by running in the background. Try this: 

`DesiredCapabilities dcaps = new DesiredCapabilities();
	dcaps.setCapability("automationName", "Windows");
	dcaps.setCapability("app", "C:\\Windows\\System32\\notepad.exe");
	dcaps.setCapability("platformName", "");
	dcaps.setCapability("deviceName", "WindowsPC");
	dcaps.getCapabilityNames();
	WindowsDriver app = new WindowsDriver<WindowsElement>(new URL("http://127.0.0.1:4723/"), dcaps);`

**Comment by bistritapcv** [comment]
Same problem with that because it changes "app" to "appium:app" and WinAppDriver is expecting "app"

**Comment by anunay1** [comment]
send a team call invite.

**Comment by bistritapcv** [comment]
Thanks.  I will.  Right now at work I have to do something else.  I will look for you sometime when you are on and when i figure out how to do that ;-)

**Comment by bistritapcv** [comment]
By the way here is the code that changes the capability in startSession()

        /**
         * Appends "appium:" prefix to all non-prefixed non-standard capabilities.
        *
        * @param possiblyInvalidCapabilities user-provided capabilities mapping.
        * @return Fixed capabilities mapping.
       */
       private static Map<String, Object> makeW3CSafe(Capabilities possiblyInvalidCapabilities) {
           return Require.nonNull("Capabilities", possiblyInvalidCapabilities)
                .asMap().entrySet().stream()
                .collect(ImmutableMap.toImmutableMap(
                    entry -> BaseOptions.toW3cName(entry.getKey()),
                    Map.Entry::getValue
                ));
      }


       /**
         * Adds the 'appium:' prefix to the given capability name if necessary.
         *
         * @param capName the original capability name.
         * @return The preformatted W3C-compatible capability name.
         */
        public static String toW3cName(String capName) {
            return W3CCapabilityKeys.INSTANCE.test(capName) ? capName : APPIUM_PREFIX + capName;
       }

      public static final String APPIUM_PREFIX = "appium:";

I can't change APPIUM_PREFIX as it is a public static final String in an unwritable file 
io.appium.java_client.internal.CapabilityHelpers

**Comment by anunay1** [comment]
Winappdriver is not W3C compliant yet.

**Comment by anunay1** [comment]
What is the selenium version that you are using?

**Comment by bistritapcv** [comment]
I don't think it matters because it is happening before Selenium, but it appears to be 4.5.0 for selenium-java

**Comment by bistritapcv** [comment]
No more thoughts?

**Comment by anunay1** [comment]
It is working perfectly fine with the pom.xml i shared.

**Comment by bistritapcv** [comment]
I don't seem to be able to find your pom

**Comment by anunay1** [comment]
Check above in comments.

**Comment by bistritapcv** [comment]
Yes, that is what I am doing.  I run WinAppDriver.exe and then run the java

**Comment by anunay1** [comment]
Send teams call request

**Comment by bistritapcv** [comment]
How do I do that?

**Comment by bistritapcv** [comment]
I will need to get permission from corporate.

**Comment by anunay1** [comment]
Ok, in the meantime try downgrading the
